tts-evaluation-MCD-DTW.ipynb

```diff
@@ -0,0 +1,501 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "191012de",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Mel params\n",
+    "n_fft=2048\n",
+    "hop_length=512\n",
+    "win_length=None\n",
+    "window='hann'\n",
+    "n_mels = 34\n",
+    "\n",
+    "## Mfcc params\n",
+    "n_mfcc=n_mels"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54830c8c",
+   "metadata": {},
+   "source": [
+    "A little bit about these parameters:  \n",
+    "    - `n_fft`: Number of fft_components or length of windowed signal after padding in stft.  \n",
+    "    - `hop_length`: Number of audio samples between adjacent STFT columns.  \n",
+    "    - `window`: Window to use in stft.  \n",
+    "    - `win_length`: Length of the window to be used.  \n",
+    "    - `n_mels`: Number of number of Mel bands to generate.\n",
+    "    - `n_mfcc`: Number of MFCCs to generate."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4cb9e37c",
+   "metadata": {},
+   "source": [
+    "### Load and Visualize data\n",
+    "Lets apply the algorithm on a synthesized, ground truth audio pair for understanding."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6efac554",
+   "metadata": {},
+   "source": [
+    "Load audio files using [librosa](https://librosa.org/doc/main/generated/librosa.load.html)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e75867c7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load files\n",
+    "gt_file = \"audio_samples/MCD-DTW/gt/sample_1.wav\"\n",
+    "synt_file = \"audio_samples/MCD-DTW/fastpitch/sample_1.wav\"\n",
+    "\n",
+    "## Load wavs\n",
+    "gt_wav, gt_sr = librosa.load(gt_file)\n",
+    "synt_wav, synt_sr = librosa.load(synt_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7aeabe75",
+   "metadata": {},
+   "source": [
+    "Generate mel specs and MFCC(Mel frequency cepstral coefficient) for ground truth and generated audio"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d1a71f88",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "## Generate spectrograms\n",
+    "gt_mels = librosa.feature.melspectrogram(gt_wav, sr=gt_sr, n_fft=n_fft, hop_length=hop_length, win_length=win_length, window=window, n_mels=n_mels)\n",
+    "synt_mels = librosa.feature.melspectrogram(synt_wav, sr=gt_sr, n_fft=n_fft, hop_length=hop_length, win_length=win_length, window=window, n_mels=n_mels)\n",
+    "## Convert Mel specs to log scale\n",
+    "gt_mels = librosa.power_to_db(gt_mels, ref=np.max)\n",
+    "synt_mels = librosa.power_to_db(synt_mels, ref=np.max)\n",
```

Comment:
When do we use the mel specs? Or the db-scaled mel specs?

Reply:
Nvm, I see that you compute MFCC from the mels, not the audio. You should use log-mel specs, not dB-mel specs

Reply:
Discussed in slack